pkg/codegen/schema/schema.go

```diff
@@ -611,17 +641,35 @@ type PropertySpec struct {
 type ObjectTypeSpec struct {
 	// Description is the description of the type, if any.
 	Description string `json:"description,omitempty"`
-	// Properties is a map from property name to PropertySpec that describes the type's properties.
+	// Properties, if present, is a map from property name to PropertySpec that describes the type's properties.
 	Properties map[string]PropertySpec `json:"properties,omitempty"`
-	// Type must be "object".
+	// Type must be "object" if this is an object type, or the underlying type for an enum.
 	Type string `json:"type,omitempty"`
-	// Requires is a list of the names of the type's required properties. These properties must be set for inputs and
-	// will always be set for outputs.
+	// Required, if present is a list of the names of an object type's required properties. These properties must be set
+	// for inputs and will always be set for outputs.
 	Required []string `json:"required,omitempty"`
 	// Language specifies additional language-specific data about the type.
 	Language map[string]json.RawMessage `json:"language,omitempty"`
 }
 
+// ComplexTypeSpec is the serializable form of an object or enum type.
+type ComplexTypeSpec struct {
+	ObjectTypeSpec
+
+	// Enum, if present, if the list of possible values for an enum type.
+	Enum []*EnumValueSpec `json:"enum,omitempty"`
+}
+
+// EnumValuesSpec is the serializable form of the values metadata associated with an enum type.
+type EnumValueSpec struct {
+	// Name, if present, overrides the name of the enum value that would usually be derived from the value.
+	Name string `json:"name,omitempty"`
+	// Description of the enum value
+	Description string `json:"description,omitempty"`
+	// Value is the enum value itself and must match an element in the Enum[] list.
```

Comment:
> must match an element in the Enum[] list.

Can you clarify? These values compose the `Enum[]` list, right?

Reply:
this is a leftover comment from when the enumMetadata was a separate object, will fix up.